comet/broker.py

```diff
@@ -570,27 +570,29 @@ def saveDataset(hash, ds, root):
 class Broker:
     """Main class to run the comet dataset broker."""
 
-    def __init__(self, data_dump_path, file_lock_time, debug, recover):
+    def __init__(self, data_dump_path, file_lock_time, debug, recover, port):
         global dumper
 
         self.config = {
             "data_dump_path": data_dump_path,
             "file_lock_time": file_lock_time,
             "debug": debug,
             "recover": recover,
+            "port": port,
         }
 
         dumper = Dumper(data_dump_path, file_lock_time)
         self.debug = debug
         self.startup_time = datetime.datetime.utcnow()
+        self.port = None
 
-    @staticmethod
-    def _wait_and_register(startup_time, config):
+    def _wait_and_register(self):
         global dumper
-        sleep(1)
-        manager = Manager("localhost", DEFAULT_PORT)
+        while not self.port:
+            sleep(1)
+        manager = Manager("localhost", self.port)
```

Comment:
What happens while `self.port = None`?

Reply:
Ah, the above is a while loop. This does not execute until `self.port` gets a while.

Reply:
Yeah, just to make sure. It's not like that will take longer than a second I hope.